lib/puppet/provider/keycloak_realm/kcadm.rb

```diff
@@ -365,6 +428,22 @@ def flush
           raise Puppet::Error, "kcadm update realms/#{resource[:name]}/default-optional-client-scopes/#{scope_id}: #{e.message}"
         end
       end
+      if @property_flush[:roles]
+        add_roles = @property_flush[:roles] - @property_hash[:roles]
```

Comment:
Move this logic up under same `if @property_flush[:roles]` so that the logic is all together.